diff --git a/Hin2n_android/app/CMakeLists.txt b/Hin2n_android/app/CMakeLists.txt
index 05510a7..965abe8 100644
--- a/Hin2n_android/app/CMakeLists.txt
+++ b/Hin2n_android/app/CMakeLists.txt
@@ -216,6 +216,7 @@ add_library(edge_v2_ipv6 SHARED
         src/main/cpp/n2n_v2_ipv6/n2n_keyfile.c
         src/main/cpp/n2n_v2_ipv6/minilzo.c
         src/main/cpp/n2n_v2_ipv6/upnp.c
+        src/main/cpp/n2n_v2_ipv6/socks5.c
         src/main/cpp/edge_jni/edge_v2_ipv6_embedded.c
         src/main/cpp/edge_jni/tuntap_android_v2_ipv6.c
         src/main/cpp/edge_jni/edge_android_v2_ipv6.c
